templates/challenge.yaml

```diff
@@ -0,0 +1,60 @@
+# These are ACL lists that correlate to actual permissions
+# on synapse
+view_acl: &view
+  - READ
+
+download_acl: &download
+  - READ
+  - DOWNLOAD
+
+edit_acl: &edit
+  - READ
+  - DOWNLOAD
+  - UPDATE
+  - CREATE
+
+edit_and_delete_acl: &edit_and_delete
+  - READ
+  - DOWNLOAD
+  - UPDATE
+  - CREATE
+  - DELETE
+
+admin_acl: &admin
+  - READ
+  - DOWNLOAD
+  - UPDATE
+  - CREATE
+  - DELETE
+  - CHANGE_SETTINGS
+  - MODERATE
+  - CHANGE_PERMISSIONS
+
+StagingProject:
+  name: Challenge Staging Example
+  type: Project
+  acl:
+    # principal_id can be both user or team id
+    - principal_id: teamA
```

Comment:
Should this be called `principal_id` if it can take principal names? Unless `teamA` is intended to be the integer ID of the team called `teamA` on Synapse. 

A related question has to do with the output JSON that's been expanded. Should we replace (or expand) any mention of a user/team with the corresponding principal IDs for stability? 

Here are some example where I replaced `principal_id` with `principal` as one way to address my first question above. I prefer expanding to include both the name and ID for readability, and I would expect SynapseFormation to use the ID over the name if both are present. 

```
# Before
    - principal: teamA
      access_type: *view

# After (replace)
    - principal: 3148189
      access_type: *view

# After (expand)
    - principal: 
        principal_id: 3148189
        principal_name: teamA
      access_type: *view
```

Reply:
Actually, a better approach is to store the expanded team node in the root to avoid duplication and use whatever method we decide on to cross-reference the team (I'm using `!Ref` here merely for illustrative purposes). The advantage of this approach is that if ever someone wants to change the team (_e.g._ to another existing team), they only need to change the ID or name in one place rather than every occurrence. 

```
# After ()
    - principal: !Ref teamA
      access_type: *view

[...]

teamA:
  principal_id: 3148189
  principal_name: teamA
```

Reply:
Oh interesting, so even if the team already exists, explicitly create the team object.  Instead of the synapse identifier that's copied, it would just be the key, and the synapse identifier can be changed over time.

Reply:
Yeah, reminiscent of database normalization using foreign keys. 

This highlights an advantage of using arrays in the root of the templates and under `children:` instead of mappings (which I prefer aesthetically), namely that `synapseformation` wouldn't have to figure out a key for each team moved to the root. Copying @mattfazza because this relates to the discussion I spurred in Sage-Bionetworks/synapseformation#27. 

I also suggest that we use `id` and `name` wherever possible (instead of variations like `principal_id`). I envision the cross-references could "look through" all of the `name`s in a template to retrieve the associated `id`. I'm not saying we should implement this now, but opting for this naming convention will make it easier for us down the road to implement cross-references. Food for 

```
# After ()

    - principal: !Ref teamA
      access_type: *view

[...]

- type: Team
  id: 3148189
  name: teamA
```